platform/darwin/src/MGLCustomVectorSource.h

```diff
@@ -0,0 +1,65 @@
+#import "MGLSource.h"
+#import "MGLGeoJSONSource.h"
+#import "MGLGeoJSONSourceBase.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@protocol MGLFeature;
+
+/**
+ Data source for `MGLCustomVectorSource`.
+ */
+@protocol MGLCustomVectorSourceDataSource <NSObject>
+
+/** 
+ Fetch data for a tile
+ @param zoom
+ @param y
+ @param x
+ @param callback A block to call with the data that has been fetched for the tile.
```

Comment:
This doesn't need to be async, since it's called on it's own NSOperationQueue. But it definitely would not work to have be it be synchronous on the main queue. Should i change this to be synchronous?